pages/api/cck/route.gpx.ts

```diff
@@ -18,7 +18,12 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
 
     mode = mode ? mode : "bicycling";
 
-    if(passcode) {
+    const daysDiff = daysSince(parseDashedDate(date as string));
```

Comment:
If `date` is not a valid 'dashed date', we should not continue. Currently, if this is the case, `daysDiff` is `NaN`, which is not greater than 1.

```
    const parsedDate = parseDashedDate(date as string)

    ...

    if(daysDiff > 1 || Number.isNaN(Number(parsedDate))) {
```

Reply:
Updated to throw error and return 400 if date cannot be successfully parsed